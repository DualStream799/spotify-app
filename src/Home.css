.Home {
  height: 100vh;
  position: fixed;
  width: 25vw;
  background: linear-gradient(#454d4b, #8f9b97);
  flex-grow: row;

  /* background-color:#8f9b97; */

  /* position:fixed  */

  /* background-color:#161616f5; */
  /* border-top-right-radius: 100px;
    border-bottom-right-radius: 100px; */
}

.player {
  width: 100%;
  height: 80px;
}


.follow_artist {
    width: 16.5vw;
    height: 56px;
    border: none;
    overflow: hidden; 
}

.lado{
    display: flex;
    flex-direction: row; 
    margin-left: 30px;
   
}

.lado2{
    display: flex;
    align-items: space-between;
    flex-direction: row; 
    margin-left: 50px;
    margin-top: 10px;
    
    
}

.bloco{
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    /* margin-left: 250px; */
    margin-left: 45vh;
    /* margin-top: -35px; */
    padding-top: 30px;
    max-width: 100%;
    width: 100%;
    color:white;
    background-color: black;
}

.bloco-favoritos{   
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    /* margin-left: 250px; */
    margin-left: 45vh;
    margin-top: -35px;
    padding-top: 30px;
    max-width: 100%;
    color:white;
    background-color: black;
    width: 87.19%;

}

.bloco3 {
  display: flex;
  align-items: center;
  font-size: 22px;
  font-weight: bold;
  margin-left: 280px;
  justify-content: center;
  font-family: Arial, Helvetica, sans-serif;
  font-weight: bold;
  padding-top: 30px;
  max-width: 100%;
  width: 100%;
  color: white;
  min-width: 100%;
  margin-top: 300px;
}

.Home {
  height: 100vh;
  position: fixed;
  width: 25vw;
  background: linear-gradient(#454d4b, #8f9b97);
  flex-grow: row;
}

.player {
  width: 100%;
  height: 80px;
}

.follow_artist {
  width: 16.5vw;
  height: 56px;
  border: none;
  overflow: hidden;
}

.lado {
  display: flex;
  flex-direction: row;
  margin-left: 30px;
}

.lado2 {
  display: flex;
  align-items: space-between;
  flex-direction: row;
  margin-left: 50px;
  margin-top: 50px;
}

.bloco {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  /* margin-left: 250px; */
  margin-left: 45vh;
  margin-top: -35px;
  padding-top: 30px;
  max-width: 100%;
  width: 100%;
  color: white;
}

.bloco-favoritos {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  /* margin-left: 250px; */
  margin-left: 45vh;
  margin-top: -35px;
  padding-top: 30px;
  max-width: 100%;
  color: white;
  width: 87.19%;
}

.form_group {
  position: relative;
  padding: 15px 0 0;
  margin-top: 10px;
  width: 50%;
  flex-direction: row;
  display: flex;
  margin-left: 30px;
}

.form_field {
  font-family: inherit;
  width: 100%;
  border: 0;
  border-bottom: 2px solid gray;
  outline: 0;
  font-size: 1.3rem;
  color: white;
  padding: 7px 0;
  background: transparent;
  transition: border-color 0.2s;
}

.bloco3 {
  display: flex;
  align-items: center;
  font-size: 22px;
  font-weight: bold;
  margin-left: 30vh;
  justify-content: center;
  font-family: Arial, Helvetica, sans-serif;
  font-weight: bold;
  padding-top: 30px;
  max-width: 100%;
  width: 100%;
  color: white;
  /* min-width: 100%; */
  margin-top: 300px;
}

.btnSearch {
  background-color: white;
  width: 120px;
  color: black;
  height: 40px;
  border-radius: 18px;
  margin-left: 20px;
  font-size: 15px;
}

.btnPlaylist {
  background-color: #454d4b;
  width: 280px;
  color: white;
  height: 40px;
  border-radius: 18px;
  margin-left: 350px;
  font-size: 15px;
  margin-top: 30px;
  margin-bottom: 10px;
}

.details-right {
  margin-left: 50vh;
}

.body_home {
  display: flex;
  justify-content: flex-start;
  height: 100vh;
  /* width: 100vh; */
}

.link {
  text-decoration: none !important;
  color: white;
}

.img {
  filter: blur(8px);
  -webkit-filter: blur(8px);
}

.botoes {
  text-decoration: none;
  text-decoration-line: none;
  text-decoration-color: #ffffff;
  display: grid;
  grid-template-columns: auto;
  grid-gap: 30px;
  z-index: 2;
  padding-top: 20px;
}

.images {
  display: flex;
  flex-direction: column;
}

.imagesRound {
  display: flex;
  border-radius: 100px;
  align-self: center;
  display: block;
  margin-left: auto;
  margin-right: auto;
}

.btn {
  background-color: transparent;
  border-color: transparent;
  color: #ffffff;
  z-index: 2;
  font-size: 18px;
  text-align: left;
  font-family: Arial, Helvetica, sans-serif;
  font-weight: bold;
}
.btn:focus {
  outline: none !important;
  outline-offset: none !important;
}

.images {
  display: flex;
  flex-direction: column;
}

.imagesRound {
  display: flex;
  /* flex-direction: column; */
  border-radius: 100px;
  align-self: center;
  display: block;
  margin-left: auto;
  margin-right: auto;
}

.btn {
  background-color: transparent;
  border-color: transparent;
  color: #ffffff;
  z-index: 2;
  font-size: 18px;
  text-align: left;
  font-family: Arial, Helvetica, sans-serif;
  font-weight: bold;
}
.btn:focus {
  outline: none !important;
  outline-offset: none !important;
}

.btnSearch:focus {
  outline: none !important;
  outline-offset: none !important;
}

.btnPlaylist:focus {
  outline: none !important;
  outline-offset: none !important;
}

.coluna {
  display: flex;
  flex-direction: column;
  margin-left: 300px;
}

.info {
  color: #ffffff;
  font-size: 30px;
  text-align: flex-start;
  font-family: "Franklin Gothic Medium", "Arial Narrow", Arial, sans-serif;
  margin-left: 30px;
  margin-bottom: -10px;
  font-weight: bold;
}

.info2 {
  color: #ffffff;
  font-size: 20px;
  text-align: flex-start;
  font-family: "Franklin Gothic Medium", "Arial Narrow", Arial, sans-serif;
  margin-left: 30px;
  margin-bottom: -15px;
}

.container {
  display: flex;
  color: white;
}

.organizacao {
  align-items: center;
  display: flex;
  flex-direction: row;
  margin-left: 50px;
}

.desc {
  display: flex;
  flex-direction: column;
  color: white;
}

.lineORetorno {
  width: 920px;
  margin: 15px;
  color: #454d4b;
}

.line {
  width: 100%;
  margin: 15px;
  color: #454d4b;
}

.centro {
  display: block;
  align-self: flex-end;
  margin-left: auto;
  margin-right: auto;
}

.title {
  color: #ffffff;
  font-size: 50px;
  font-family: Arial, Helvetica, sans-serif;
  margin-left: 20px;
}

.escutados {
  align-items: flex-start;
  display: flex;
  flex: 1;
  flex-direction: column;
  color: white;
  min-width: 100%;
  width: 100%;
}

.antigas {
  margin-left: 30px;
  font-family: Arial, Helvetica, sans-serif;
  font-size: 18px;
  flex-direction: row;
  display: flex;
  justify-content: space-between;
  margin-right: 50px;
  color: rgb(255, 254, 254);
  width: 800px;
  background-color: rgb(0, 0, 0);
}

.separando {
  flex-direction: row;
  display: flex;
  width: 84%;
  padding-left: 25px;
  color: white;
  justify-content: space-between;
  font-size: 24px;
  margin-bottom: 20px;
  font-weight: bold;
}

.separando_antigas {
  flex-direction: row;
  display: flex;
  width: 74%;
  padding-left: 25px;
  color: white;
  justify-content: space-between;
  font-size: 26px;
  margin-bottom: 20px;
  font-weight: bold;

}

.grid-container {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  align-items: center;
}

.tracks-container {

    display: grid;
    grid-template-columns: 1fr 1fr 1fr; 
    align-items: left;
    width: 74rem;
  }

.musicas{
    flex: 1;
    
    display: flex;
    flex-direction: row;
    justify-content: space-around;
    color:white;
    
}

.grid-item{
    margin: 1vw;
    flex-direction: row;
    color:white;

}

.musicas {
  flex: 1;

  display: flex;
  flex-direction: row;
  justify-content: space-around;
  color: white;
}

.grid-item {
  margin: 1vw;
  flex-direction: row;
  color: white;
}

.TrackName {
  font-weight: bold;
  font-size: 25px;
  margin-bottom: -10px;
  color: white;
}

.containerEscutando {
  display: flex;
  flex-direction: column;
}

.Texto {
  color: white;
}

.centralizacao {
  text-align: center;
}

.blocoEscutando {
  display: flex;
  width: 75vw;
  flex-direction: column;
  align-items: center;
  width: 100vh;
  margin-left: 42px;
}

.musicatxt {
  color: white;
  font-size: 25px;
  font-weight: bold;
  align-self: center;
}


.antigastxt{
    flex-grow: row;
    width: 24rem;
    margin-left: 2rem;
    margin-right: 2rem;
}

.albumtxt {
  color: white;
  font-size: 20px;
  display: flex;
  align-self: center;
  margin-top: -30px;
}

.containerGeral {
  width: 100%;
  margin-left: 250px;
  /* flex:1; */
  /* width:100vw;
    margin-left: 25vw; */
  /* flex:1; */
  display: flex;
  flex-direction: column;
  height: 100%;
  /* min-height: 100%; */
  /* min-width: 100%; */
}

.descricao {
  margin-left: 30px;
  font-size: 18px;
}

.input {
  display: flex;
  height: 40px;
  width: 200px;
  align-items: center;
  border-radius: 20px;
  justify-content: center;
  margin-left: 30px;
  background-color: white;
  margin-bottom: 10px;
}

.lista {
  display: grid;
  grid-template-columns: 2fr 2fr;
}

.artistaTitulo {
  margin-left: 30px;
  font-size: 30px;
}

.inputGenero {
  display: flex;
  height: 40px;
  width: 200px;
  align-items: center;
  border-radius: 20px;
  justify-content: center;
  margin-left: 30px;
  background-color: white;
  margin-bottom: 10px;
}
.generos {
  display: grid;
  grid-template-columns: 4fr 4fr 4fr;
  margin-left: 130px;
}

.blocoFavoritos {
  margin-left: 30px;
}

.testeCheck {
  color: black;
}

.btnSearch {
  align-self: center;
  margin-top: 30px;
  margin-left: 105px;
  font-weight: bold;
  height: 40px;
  color: "#454D4B";
}

.blocoHome {
  align-items: center;
  justify-content: center;
  display: flex;
  margin-left: 480px;
  margin-top: 150px;
  flex-direction: column;
}

label,
label::before {
  -webkit-transition: 0.25s all ease;
  -o-transition: 0.25s all ease;
  transition: 0.25s all ease;
}

input[type="checkbox"]:checked + label,
input[type="radio"]:checked + label {
  padding-left: 1em;
  color: black;
}

.txt {
  display: flex;
  align-items: center;
  flex-direction: column;

  color: rgb(253, 246, 246);
  margin-top: 60px;
  font-size: 22px;
}

.bemVindo {
  font-weight: bold;
  font-size: 24px;
  margin-bottom: -16px;
}

.dark-mode {
  background-color: black;
}

.light-mode {
  background-color: #9da7a4;
}

:root {
  --main-bg-color: rgb(27, 59, 38);
}
