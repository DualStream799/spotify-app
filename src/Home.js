import { BrowserRouter, Switch, Route } from 'react-router-dom';
import NewReleases from './components/NewReleases';
import App from './App'
import Recomended from './components/Recomended'
import Favorites from './components/Favorites';
import Nav from './components/Nav' 
import UserPlaylist from './components/UserPlaylist'
import { Component } from 'react';
import axios from 'axios';

class Home extends Component {
	constructor(props) {
		super(props);
		const parametros = this.getHashParams();
		this.token = parametros.access_token;
		localStorage.setItem('token', this.token)

		this.state = {
		};
	}

	componentDidMount= () => {
		axios
			.get('https://api.spotify.com/v1/me/tracks?limit=30', {
				headers: {
					Authorization: `Bearer ${this.token}`
				}
			})
			.then((response) => {
				this.setState({
					dataTracks: response.data.items,
					data: response.data.items,
				});
				console.log(this.state.data)
			})
			.catch((erro) => console.log(erro.response.data));
	};


	getHashParams = () => {
		var hashParams = {};
		var e,
			r = /([^&;=]+)=?([^&;]*)/g,
			q = window.location.hash.substring(1);
		e = r.exec(q);
		while (e) {
			hashParams[e[1]] = decodeURIComponent(e[2]);
			e = r.exec(q);
		}
		return hashParams;
	}
	render() {
	return(
		<BrowserRouter>
			<div>
				<Nav />
				<Switch>
					<Route exact path='/' component={App}/>
					<Route exact path='/newReleases' component={NewReleases}/>
					<Route exact path='/recomended' component={Recomended}/>
					<Route exact path='/Favorites' component={Favorites}/>
					<Route exact path='/userPlaylist' component={UserPlaylist}/>
				</Switch>
			</div>
		</BrowserRouter>
	)}
}

export default Home;