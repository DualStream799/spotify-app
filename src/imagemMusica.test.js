import React from 'react';
import Home from './Home';
import { shallow, mount } from 'enzyme';
import toJson from "enzyme-to-json"
import Select from '@material-ui/core/Select';

it("should render imagens das músicas em Minhas Músicas", () => {
    const img = shallow(<img data-testid='musicImg'/>)
    expect(img).toMatchSnapshot()
})
