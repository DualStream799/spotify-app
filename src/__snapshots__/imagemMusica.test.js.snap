// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render imagens das músicas em Minhas Músicas 1`] = `ShallowWrapper {}`;
